File: LeetCode-Programs/Reverse Linked List.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
       ListNode *temp=NULL, *nextNode = NULL;
        while(head!=NULL)
        {
            nextNode=head->next;
            head->next=temp;
            temp=head;
            head=nextNode;
        }
        return temp;
    }
};

File: LeetCode-Programs/Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp=None
        NextNode=None
        while head!=None:
            NextNode=head.next
            head.next=temp
            temp=head
            head=NextNode
        return temp
            
